plutus-chain-index/src/Plutus/ChainIndex/Types.hs

```diff
@@ -55,12 +49,27 @@ pageOf (PageSize ps) items =
 
 -- | The tip of the chain index
 data Tip =
-    -- TODO: Add genesis block tip
-    Tip
+      TipAtGenesis
+    | Tip
         { tipSlot    :: Slot -- ^ Last slot
         , tipBlockId :: BlockId -- ^ Last block ID
         , tipBlockNo :: Int -- ^ Last block number
         }
-    deriving stock (Eq, Ord, Show, Generic)
+    deriving stock (Eq, Show, Generic)
     deriving anyclass (ToJSON, FromJSON)
 
+-- | This mirrors the previously defined Tip which used the Last monoid definition.
+--   However, I am not sure what the meaning of Tip being a monoid really is. What
```

Comment:
We needed the `Last Tip` so that we could derive a `Monoid` instance for `UtxoState`. We always have to append newer blocks to the right anyway (`<>` for `TxUtxoBalance` is not commutative). I agree that there doesn't seem to be a good `Monoid` for `Tip` so I think we should stick with `Last Tip` and drop the instance here.

Reply:
I think we should keep this monoid instance, and mention in the comment that it is not commutative.